src/wasmv1_execution/abi/abis.rs

```diff
@@ -863,7 +938,37 @@ fn abi_send_async_message(
                 &req.data,
                 filter,
             ) {
-                Ok(_) => resp_ok!(SendAsyncMessageResult, {}),
+                Ok(_) => {
```

Comment:
this is as long as the original function
I'd put it in a helper functions on the exec_env
and call
```rust
exec_env.push_trace(function_name!(), <params inlined>);
```
with 
```rust
#[cfg(feature ="execution-trace")]
fn push_trace(nf_name: &str, params: Vec<>) {
	// handle all the check here
}

#[not cfg(feature ="execution-trace")]
fn push_trace(nf_name: &str, params: Vec<>) {
	// empty
}

```
it should make the call site cleaner

Reply:
As we now need the parameter name this is not an option anymore